databricks/koalas/indexes.py

```diff
@@ -399,6 +400,70 @@ def copy(self, name=None):
             result.name = name
         return result
 
+    def symmetric_difference(self, other, result_name=None, sort=None):
+        """
+        Compute the symmetric difference of two Index objects.
+
+        Parameters
+        ----------
+        other : Index or array-like
+        result_name : str
+        sort : True or None, default None
+            Whether to sort the resulting index.
+            * True : Attempt to sort the result.
+            * None : Do not sort the result.
+
+        Returns
+        -------
+        symmetric_difference : Index
+
+        Notes
+        -----
+        ``symmetric_difference`` contains elements that appear in either
+        ``idx1`` or ``idx2`` but not both. Equivalent to the Index created by
+        ``idx1.difference(idx2) | idx2.difference(idx1)`` with duplicates
+        dropped.
+
+        Examples
+        --------
+        >>> s1 = ks.Series([1, 2, 3, 4], index=[1, 2, 3, 4])
+        >>> s2 = ks.Series([1, 2, 3, 4], index=[2, 3, 4, 5])
+
+        >>> s1.index.symmetric_difference(s2.index)
+        Int64Index([5, 1], dtype='int64')
+
+        You can set name of result Index.
+
+        >>> s1.index.symmetric_difference(s2.index, result_name='koalas')
+        Int64Index([5, 1], dtype='int64', name='koalas')
+
+        You can set sort to `True`, if you want to sort the resulting index.
+
+        >>> s1.index.symmetric_difference(s2.index, sort=True)
+        Int64Index([1, 5], dtype='int64')
+
+        You can also use the ``^`` operator:
+
+        >>> s1.index ^ s2.index
+        Int64Index([5, 1], dtype='int64')
+        """
+        sdf_self = self._kdf._sdf.select(self._internal.index_scols)
+        sdf_other = other._kdf._sdf.select(other._internal.index_scols)
+
+        sdf_symdiff = sdf_self.union(sdf_other) \
+                              .subtract(sdf_self.intersect(sdf_other))
+
+        if sort:
+            sdf_symdiff = sdf_symdiff.sort(self._internal.index_scols)
+
+        internal = self._kdf._internal.copy(sdf=sdf_symdiff)
+        result = Index(DataFrame(internal))
```

Comment:
This should be `DataFrame(internal).index`? E.g., what if the index is multi-index?